### ISSUE TITLE
Allow (formatter?) config to never introduce implicit concatted strings or just accept that a check run will complain about it

### ISSUE THREAD

**Issue by jankatins** [astral-sh/ruff#14638]
I think implicitly concat'ed strings are pure evil since a time where I had to debug an accidentally added comma in a place where `+` would have been proper, which made a single string a tuple of two strings. I would like ruff to allow this and not complain about this, but currently it does complain and IMO it complains for the wrong reasons and with the wrong solution: 

I don't want it to stop complaining about implicitly concat'ed strings, but stop the formatter from introducing them by using a '+' between multiline strings. Or if it introduces implicitly concat'ed strings (because + would be a black vialoation), just complain (when it happens), so I'm forced to fix it by hand (which I would be fine with). But having the scary warning in all runs makes me and especially unwary users scared into disabling that rule. :-(

Inspired by https://github.com/astral-sh/ruff/issues/13031#issuecomment-2302402183, I would have expected that

```toml
[tool.ruff.lint]
extend-select = ["ISC"]
ignore=["ISC003"] # Don't complain about explicitly concat'ed strings
unfixable = ["ISC001"] # don't fix one line implicitly concat'ed strings, but just complain and make the user fix it

[tool.ruff.lint.flake8-implicit-str-concat]
allow-multiline = false
```

would work, but ruff still complains, that this is incompatible with the formatter:

```
 The following rule may cause conflicts when used with the formatter: `ISC001`. To avoid unexpected behavior, we recommend disabling this rule, either by removing it from the `select` or `extend-select` configuration, or adding it to the `ignore` configuration.
```

**Comment by MichaReiser** [comment]
I think what you're asking for will be addressed with the upcoming #13371 

* The formatter will merge implicitly concatenated strings that fit on a single line into a single string literal (there are a few edge cases where that's not possible but they are very rare)
* The formatter will never collapse an implicit concatenated string https://github.com/astral-sh/ruff/pull/13928

These two changes will make the formatter compatible with ISC001 (also see https://github.com/astral-sh/ruff/issues/8272).

**Comment by MichaReiser** [comment]
> would work, but ruff still complains, that this is incompatible with the formatter:

This might be a bug but I'm not sure if it's still worth fixing considering that the new style guide will be released shortly. But feel free to submit a PR that suppresses the warning if that setting is set.

**Comment by jankatins** [comment]
> The formatter will merge implicitly concatenated strings that fit on a single line into a single string literal (there are a few edge cases where that's not possible but they are very rare)

This is actually something I don't want: up to now, all such cases of mine were bad coding form my side and I in a lot of cases I had to pick a comma instead of a  plus. Just merging the strings will mean that `["A", "B" "C"]` will end up as `["A", "BC"]` which usually is a bug in my case.

**Comment by MichaReiser** [comment]
I see. So the problem here is that you introduce implicit concatenated strings and you want the formatter to leave them unchanged? 

I don't think we should introduce an option for this because it is in direct conflict with consistent formatting. However, I do see how the formatter joining the strings can be "annoying" in an editor context. IMO, the benefit of the formatter joining the strings is that it becomes very apparent that you forgot a `,` and hitting Ctrl + Z in the editor should undo the formatter change.





> but stop the formatter from introducing them by using a '+' between multiline strings. Or if the formatter needs to introduce implicitly concat'ed strings (because + would be a black violation), 

The formatter never introduces implicit concatenated strings. It only formats implicit concatenated strings that were already present in the source.
